src/Addons/CategoryFetcher.php

```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\ProductComment\Addons;
+
+use PrestaShop\CircuitBreaker\AdvancedCircuitBreakerFactory;
+use PrestaShop\CircuitBreaker\Contracts\Factory;
+use PrestaShop\CircuitBreaker\Storages\SymfonyCache;
+use Symfony\Component\Cache\Simple\FilesystemCache;
+use Symfony\Component\CssSelector\CssSelectorConverter;
+use DOMDocument;
+use DOMXPath;
+use DOMNode;
+
+/**
+ * Class CategoryFetcher helps you to fetch an Addon category data. It calls the Addons
+ * API for name and link, and scrap the Addons platform to get its description.
+ * Every call is protected by a CircuitBreaker to avoid blocking the back-office.
+ */
+class CategoryFetcher
+{
+    const ADDONS_BASE_URL = 'https://addons.prestashop.com';
+    const ADDONS_API_URL = 'https://api-addons.prestashop.com';
+
+    const CLOSED_ALLOWED_FAILURES = 2;
+    const API_TIMEOUT_SECONDS = 0.6;
```

Comment:
600ms is really fast :thinking:  
What if we have a lot of data?

Reply:
With my "awesome" connection I had responses between 100 and 300ms. So I double the longest to define 600ms. But I can increase it if you think it's safer? 1sec maybe?

Reply:
If you're using an edge connection / 3G you can download at 10ko/s :trollface: 
So yes, increase it to 1s or 2s could be better.

Reply:
Yes but you're referring to your connection while using the website locally
This code is executed server-side so the server's connection will be used, I doubt there are many server which use 3G connection

Reply:
Even on never side. What if there's a lot a people on addons and it tooks 1 / 2 second to get a response :/

Reply:
then you won't have the response right now, the circuit breaker goes to OPEN state and the english fallback is used, then once the threshold is expired (60secs for now for test purpose but I will increase it for final release) the circuit breaker reopens and fetch the response caching it for 24h
I think this behaviour is acceptable, wdyt?